rsmtool/utils/commandline.py

```diff
@@ -230,117 +253,620 @@ def existing_configuration_file(string):
+        - "format" : a field that accepts possible file formats (csv/tsv/xlsx)
+        - "id" : a field that accepts experiment/comarison/summary IDs
+        - "integer" : a field that accepts integer values
+        - "text" : a field that accepts open-ended text
+    label : str
+        The label for the field that will be displayed to the user.
+    prompt_method : callable
+        The function that will be used to compute the value for the field.
+        The main difference arises between field that accept only a single
+        value vs. multiple values.
+    validator : prompt_toolkit.validation.Validator
+        A ``Validator`` object used to validate the values for the field
+        as they are entered by the user. Just like ``completer``, the
+        type of ``Validator`` used depends on the ``data_type`` of the field.
+        For example,
+    """
+
+    def __init__(self, field_name, field_type, field_metadata):
+        """
+        Create a new InteractiveField instance for the given field name
+        and with the given field type (required/optional).
+
+        Parameters
+        ----------
+        field_name : str
+            The internal name of the field as used in the configuration
+            dictionary.
+        field_type : str
+            One of "required" or "optional", depending on whether the
+            field is required or optional in the configuration.
+        field_metadata : dict
+            A dictionary containing the pre-defined metadata attributes
+            for the given field. This dictionary is required to have
+            the "label" key and can have the following optional
+            keys: "choices", "count", and "type". For descriptions of what
+            these keys mean, see the docstring for the ``InteractiveField``
+            class. Examples of such dictionaries can be found in ``rsmtool.utils.constants.INTERACTIVE_MODE_METADATA``.
+        """
+        # assign metadata attributes to class attributes
+        self.field_name = field_name
+        self.field_type = field_type
+        self.label = field_metadata['label']
+        self.choices = field_metadata.get('choices', [])
+        self.count = field_metadata.get('count', 'single')
+        self.data_type = field_metadata.get('type', 'text')
+
+        # instantiate the interaction-related attributes to their default values
+        self.completer = None
+        self.complete_style = None
+        self.validator = None
+
+        # now override these attributes as necessary depending on field data types
+        if self.data_type == 'boolean':
+            allow_empty = field_type == 'optional'
+            self.completer = WordCompleter(['true', 'false'])
+            self.validator = self._make_boolean_validator(allow_empty=allow_empty)
+        elif self.data_type == 'choice':
+            if not self.choices:
+                raise(ValueError, f"invalid list of choices for {field_name}")
```

Comment:
I think this results in an exception:

```
In [1]: raise(ValueError, "hello")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-711d55f1853f> in <module>
----> 1 raise(ValueError, "hello")

TypeError: exceptions must derive from BaseException
```

```suggestion
                raise ValueError(f"invalid list of choices for {field_name}")
```

Reply:
Nice catch!